app/models/roster_entry.rb

```diff
@@ -10,6 +13,27 @@ class IdentifierCreationError < StandardError; end
 
   before_create :validate_identifiers_are_unique_to_roster
 
+  scope :order_by_repo_created_at, lambda { |context|
+    assignment = context[:assignment]
+    sql_formatted_assignment_id = assignment.id
+
+    order("assignment_repos.created_at")
+      .joins <<~SQL
+        LEFT OUTER JOIN assignment_repos
+        ON roster_entries.user_id = assignment_repos.user_id
+        AND assignment_repos.assignment_id=#{sql_formatted_assignment_id}
```

Comment:
Can you replace this string interpolation with SQLs `?` interpolation?

Reply:
Unfortunately multi-line SQL statements in this format `?` is not possible -- I followed the same format that the rest of this file does for raw SQL queries (of which some also use string interpolation), but I can change it if you like.

Reply:
Relaying from Slack, looks like we can't do ? interpolation into `joins` with AR. This is fine